bin/test.js

```diff
@@ -126,44 +138,34 @@ export class WTRConfig {
 	}
 
 	#getPattern(type) {
-		const pattern = this.#cliArgs.files || this.pattern(type);
+		const pattern = [].concat(this.#cliArgs.files || this.pattern(type));
 
 		// replace filename wildcards with all filter strings
 		// e.g. If filter is ['button', 'list'], pattern './test/*.test.*' becomes:
 		// [ './test/+(*button*.test.*|*.test.*button*)', './test/+(*list*.test.*|*.test.*list*)' ]
 		if (this.#cliArgs.filter) {
 			return this.#cliArgs.filter.map(filterStr => {
 				// replace everything after the last forward slash
-				return pattern.replace(/[^/]*$/, fileGlob => {
+				return pattern.map(p => p.replace(/[^/]*$/, fileGlob => {
 					// create a new glob for each wildcard
 					const fileGlobs = Array.from(fileGlob.matchAll(/(?<!\*)\*(?!\*)/g)).map(({ index }) => {
 						const arr = fileGlob.split('');
-						arr.splice(index, 1, `*${filterStr}*`);
+						arr.splice(index, 1, filterStr);
 						return arr.join('');
 					});
-					return `+(${fileGlobs.join('|')})`;
-				});
-			});
+					return `+(${fileGlobs.join('|') || fileGlob})`;
```

Comment:
Handles bad POSIX CLI globs gracefully.

`--files ./*.test.js` is converted by the OS to a list of all matching files (with no * characters) before being sent to the process.
`--files './*.test.js'` is what _should_ be used. We can add standard POSIX glob note to docs.